src/nvim/ex_docmd.c

```diff
@@ -5640,6 +5641,20 @@ static void ex_syncbind(exarg_T *eap)
   }
 }
 
+void do_read_cmd(exarg_T *eap)
+{
+  char *cmd = eap->arg + 1;
+  StringBuilder put_cmd = KV_INITIAL_VALUE;
+  kv_printf(put_cmd, "try | ");
```

Comment:
wrapping things in a `try` means that if there is an error with the command no text is inserted into the buffer. 

This _feels_ like the right behaviour to me, but potentially could someone want the error text inserted? When using `read !cmd` text will be inserted into the buffer even if there is an error.